src/renderer/atlas/BackendD3D.cpp

```diff
@@ -1124,7 +1143,30 @@ void BackendD3D::_drawTextOverlapSplit(const RenderingPayload& p, u16 y)
     }
 }
 
-bool BackendD3D::_drawGlyph(const RenderingPayload& p, const BackendD3D::AtlasFontFaceEntryInner& fontFaceEntry, BackendD3D::AtlasGlyphEntry& glyphEntry)
+void BackendD3D::_initializeFontFaceEntry(AtlasFontFaceEntryInner& fontFaceEntry)
+{
+    ALLOW_UNINITIALIZED_BEGIN
+    std::array<u32, 0x100> codepoints;
+    std::array<u16, 0x100> indices;
+    ALLOW_UNINITIALIZED_END
+
+    for (u32 i = 0; i < codepoints.size(); ++i)
+    {
+        codepoints[i] = 0x2500 + i;
+    }
+
+    THROW_IF_FAILED(fontFaceEntry.fontFace->GetGlyphIndicesW(codepoints.data(), codepoints.size(), indices.data()));

```

Comment:
so I understand: this function caches every glyph ID from the font that mapped to a box drawing character?

Reply:
Yep exactly. The alternative is to check the codepoints during text shaping, but I felt like that would be just as bad for code complexity. This approach has the additional benefit that it only needs to run the box-glyph specific logic once, when the glyph is rasterized, and not every time the text is shaped. Just like the alternative approach (checking codepoints) it only works for basic cmap mappings though (so no substitutions, etc.).